.github/workflows/pull-request.yaml

```diff
@@ -1,11 +1,212 @@
 name: Test Changes
 
-on: [pull_request]
+on:
+  pull_request:
+    paths-ignore:
+      - '*.md'
+
+  push:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+.x
+
+env:
+  GRADLE_OPTS: '-Dorg.gradle.daemon=false -Dkotlin.compiler.execution.strategy=in-process'
 
 jobs:
   build:
     runs-on: ubuntu-18.04
 
     steps:
-      - name: Dummy job
-        run: echo "Hello world"
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Cache gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+            ~/.konan
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Install Java 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
```

Comment:
I've found it's useful to define dependency versions at the top of the file, so that it's a single change when you need to change versions.

For example:
```
env:
    JAVA_VERSION: 11

...

- name: Install Java ${{ env.JAVA_VERSION }}
    uses: actions/setup-java@v1
    with:
      java-version: ${{ env.JAVA_VERSION }}
```

Reply:
Looking through the rest of the file, you could use this pattern for protobuf, node, and java.